src/ranch_server_proxy.erl

```diff
@@ -0,0 +1,48 @@
+-module(ranch_server_proxy).
+
+-behavior(gen_server).
+
+-export([start_link/0]).
+-export([init/1]).
+-export([handle_call/3]).
+-export([handle_cast/2]).
+-export([handle_info/2]).
+-export([code_change/3]).
+
+start_link() ->
+	gen_server:start_link(?MODULE, [], []).
+
+init([]) ->
+	case wait_ranch_server(50) of
+		{ok, Monitor} ->
+			{ok, Monitor, hibernate};
+		{error, Reason} ->
+			{stop, Reason}
+	end.
+
+handle_call(_, _, Monitor) ->
+	{noreply, Monitor, hibernate}.
+
+handle_cast(_, Monitor) ->
+	{noreply, Monitor, hibernate}.
+
+handle_info({'DOWN', Monitor, process, _, Reason}, Monitor) ->
+	{stop, Reason, Monitor};
+handle_info(_, Monitor) ->
+	{noreply, Monitor, hibernate}.
+
+code_change(_, Monitor, _) ->
+	{ok, Monitor}.
+
+-spec wait_ranch_server(non_neg_integer()) -> {ok, reference()} | {error, term()}.
```

Comment:
This spec on the other hand is probably not necessary.

Reply:
Removed.